### PR TITLE
Rem 291

### PR DESCRIPTION
**Pull request by alihajjhassan**

### CODE REVIEW

File: src/pages/pos-users/index.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    invalidate()
+    toast.success('Record Deleted Successfully')
+  }
+
+  function openForm(userId) {
+    stack({
+      Component: PosUsersForm,
+      props: {
+        labels: labels,
+
```
Comment (RebeccaFrem02) [inline src/pages/pos-users/index.js:86]: remove unused empty lines

**Comment by FatimaSalami** [comment]
Info button behaviour is not correct: on save it should be enabled.. and on edit mode when clear button is clicked it should be disabled.

File: src/pages/pos-users/index.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        userId: userId ? userId : null,
+        maxAccess: access,
+        invalidate: invalidate
```
Comment (hadichahine) [inline src/pages/pos-users/index.js:89]: Call useINvalidate inside the form.

File: src/pages/pos-users/index.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  const popup = obj => {
+    openForm(obj?.userId)
+  }
+
+  const del = async obj => {
+    await postRequest({
+      extension: PointofSaleRepository.PosUsers.del,
+      record: JSON.stringify(obj)
+    })
+    invalidate()
```
Comment (hadichahine) [inline src/pages/pos-users/index.js:77]: refetch or invalidate can work. try refetch

File: src/pages/pos-users/forms/PosUsersForm.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        }
+      } catch (e) {}
+    })()
+  }, [])
+
+  return (
+    <FormShell resourceId={ResourceIds.PriceLevels} form={formik} maxAccess={maxAccess} editMode={editMode}>
+      <VertLayout>
+        <Grow>
+          <Grid container spacing={4}>
```
Comment (hadichahine) [inline src/pages/pos-users/forms/PosUsersForm.js:68]: Do we need all those grids?

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/pos-users/forms/PosUsersForm.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                  valueField='recordId'
+                  parameters={`_size=1000&_filter=&_startAt=0&_sortBy=fullName`}
+                  displayField={['email']}
+                  columnsInDropDown={[{ key: 'email', value: 'email' }]}
+                  values={formik.values}
+                  required
+                  maxAccess={maxAccess}
+                  readOnly={editMode}
+                  onChange={(event, newValue) => {
+                    formik.setFieldValue('userId', newValue ? newValue.recordId : '')
```
Comment (hadichahine) [inline src/pages/pos-users/forms/PosUsersForm.js:84]: newValue?.recordId works too

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/pos-users/index.js (hunk 2 above)
Comment (alihajjhassan) [reply src/pages/pos-users/index.js:89]: i tried doing that but didnt work

**Pull request merged**